internal/controlplane/handlers_repositories.go

```diff
@@ -98,18 +98,13 @@ func (s *Server) ListRepositories(ctx context.Context,
 	entityCtx := engine.EntityFromContext(ctx)
 	projectID := entityCtx.Project.ID
 
-	provider, err := getProviderFromRequestOrDefault(ctx, s.store, in, projectID)
-	if err != nil {
-		return nil, providerError(err)
-	}
-
 	reqRepoCursor, err := cursorutil.NewRepoCursor(in.GetCursor())
 	if err != nil {
 		return nil, util.UserVisibleError(codes.InvalidArgument, err.Error())
 	}
 
 	repoId := sql.NullInt64{}
-	if reqRepoCursor.ProjectId == projectID.String() && reqRepoCursor.Provider == provider.Name {
```

Comment:
Why are we removing the provider from the repo cursor?

Reply:
If we don't get the provider from the context, then we can't compare it to what we have in the cursor

Reply:
See my next comment -- I think the cursor could contain `""` or an actual provider name, since this is ListRepositories.